cmd/query/app/server_test.go

```diff
@@ -74,7 +76,7 @@ func TestServerBadHostPort(t *testing.T) {
 
 func TestServerInUseHostPort(t *testing.T) {
 
-	for _, hostPort := range [2]string{":8080", ":8081"} {
+	for _, hostPort := range [2]string{"127.0.0.1:8080", "127.0.0.1:8081"} {
```

Comment:
a cleaner way to write tests is to listen on port 0 and let the OS assign the unused port. Can we do it here?

Reply:
I believe the goal of this test is to intentionally have two listeners listening on the same host:port to cause a "Bind address already in use" error, and so there's a need for the port numbers to be deterministic rather than dynamically allocated by the OS.

Please let me know if I've misunderstood your question.

Reply:
I haven't read the test carefully, but from it's name it's trying to test for port conflict, which does not require listening on a _specific_ port, it just requires trying to bind twice to _the same port_. The first time it can bind to a random port and use it for the second attempt.

Reply:
That's neat! I didn't think it was possible to retrieve the OS allocated port. Thanks!